src/EPR.Calculator.Service.Function/Services/CalculatorRunService.cs

```diff
@@ -1,56 +1,177 @@
+        /// <param name="args">The calculator run parameters.</param>
+        /// <param name="pipelineName">The name of the pipeline.</param>
+        /// <returns>The Azure Synapse runner parameters.</returns>
+        public static AzureSynapseRunnerParameters GetAzureSynapseConfiguration(
+            CalculatorRunParameter args,
+            string pipelineName)
+        {
+            var financialYear = args.FinancialYear;
+            return new AzureSynapseRunnerParameters
+            {
+                PipelineUrl = new Uri(Configuration.PipelineUrl),
+                CheckInterval = int.Parse(Configuration.CheckInterval),
+                MaxCheckCount = int.Parse(Configuration.MaxCheckCount),
+                PipelineName = pipelineName,
+                CalculatorRunId = args.Id,
+                FinancialYear = Common.Utils.Util.GetCalendarYearFromFinancialYear(financialYear),
+            };
+        }
+
+        /// <summary>
+        /// Builds the JSON content of the status update API call.
+        /// </summary>
+        /// <param name="calculatorRunId">The calculator run ID.</param>
+        /// <param name="pipelineSucceeded">Indicates whether the pipeline succeeded.</param>
+        /// <returns>The JSON content for the status update.</returns>
+        public static StringContent GetStatusUpdateMessage(int calculatorRunId, bool pipelineSucceeded)
+        {
+            var statusUpdate = new
+            {
+                runId = calculatorRunId,
+                updatedBy = "string",
+                isSuccessful = pipelineSucceeded,
+            };
+
+            return new StringContent(
+                JsonSerializer.Serialize(statusUpdate),
+                Encoding.UTF8,
+                "application/json");
+        }
+
+        /// <summary>
+        /// Creates a JSON message containing the calculator run ID for preparing calculation results.
+        /// </summary>
+        /// <param name="calculatorRunId">The ID of the calculator run.</param>
+        /// <returns>A <see cref="StringContent"/> object containing the JSON message.</returns>
+        public static StringContent GetPrepareCalcResultMessage(int calculatorRunId)
+        {
+            var calcResultsRequest = new
+            {
+                runId = calculatorRunId,
+            };
+
+            return new StringContent(
+                JsonSerializer.Serialize(calcResultsRequest),
+                Encoding.UTF8,
+                "application/json");
+        }
+
+        /// <summary>
+        /// Starts the calculator process.
+        /// </summary>
+        /// <param name="calculatorRunParameter">The parameters required to run the calculator.</param>
+        /// <returns>
+        /// A task that represents the asynchronous operation. The task result contains a boolean indicating success or failure.
+        /// </returns>
+        public async Task<bool> StartProcess(CalculatorRunParameter calculatorRunParameter)
```

Comment:
This is a very long method - might be good to either break it into several smaller ones, or add some comments to highlight the start of different sections of it at a glance.

Reply:
As suggested refactored the code